sim/midas/targetutils/src/main/scala/midas/annotations.scala

```diff
@@ -343,4 +343,21 @@ object TriggerSink {
       def toFirrtl = TriggerSinkAnnotation(targetWire.toTarget, clock.toTarget)
     })
   }
+
+  /**
+    * Syntatic sugar for a when context that is predicated by a trigger sink.
+    * Example usage:
+    * {{{
+    * TriggerSink.whenEnabled {
```

Comment:
I'm guessing this is sugar to enable the functionality while not embedding the trigger in the printf transform?

Reply:
Yeah, you don't need to use this function, but it serves as shorthand for how i expected users should. 

We could couple printfs and triggers, as we do for autocounter. That might be a question for a wider discussion. I opted not to because it was simpler in the short run, and keeping things decoupled makes it easier to test these systems.